### PR TITLE
media-libs/libsdl: multiple CVEs + EAPI 7

### PR DESCRIPTION
**Pull request by vaukai**

### CODE REVIEW

**Comment by ajakk** [comment]
Any chance you could get the other patches in the security bug to apply?

**Comment by vaukai** [comment]
> Any chance you could get the other patches in the security bug to apply?

Done (hopefully). Tried to apply them in same order as found in git log from "git clone https://github.com/libsdl-org/SDL-1.2"
Could you please help and cross-check if I got them all?

File: media-libs/libsdl/libsdl-1.2.15_p20210224.ebuild [hunk 1]
```diff
@@ -0,0 +1,134 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+inherit autotools flag-o-matic multilib toolchain-funcs multilib-minimal
```
Comment (a17r) [inline media-libs/libsdl/libsdl-1.2.15_p20210224.ebuild:5]: empty line after EAPI please

File: media-libs/libsdl/libsdl-1.2.15_p20210224.ebuild [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+		ewarn "Since you've chosen to use possibly unsafe CFLAGS,"
+		ewarn "don't bother filing libsdl-related bugs until trying to remerge"
+		ewarn "libsdl without the custom-cflags use flag in USE."
+	fi
+}
+
+PATCHES=(
+		"${FILESDIR}"/${PN}-$(ver_cut 1-3)-sdl-config.patch
+#		"${FILESDIR}"/${PN}-$(ver_cut 1-3)-resizing.patch					# Still needed?
+		"${FILESDIR}"/${PN}-$(ver_cut 1-3)-gamma.patch
```
Comment (a17r) [inline media-libs/libsdl/libsdl-1.2.15_p20210224.ebuild:60]: too deep indendation

File: media-libs/libsdl/libsdl-1.2.15_p20210224.ebuild (hunk 1 above)
Comment (a17r) [inline media-libs/libsdl/libsdl-1.2.15_p20210224.ebuild:5]: I don't see `multilib` nor `toolchain-funcs` in use here, can you confirm?

File: media-libs/libsdl/libsdl-1.2.15_p20210224.ebuild [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+HOMEPAGE="https://libsdl.org/"
+SRC_URI="https://github.com/libsdl-org/SDL-1.2/archive/${MY_COMMIT}.tar.gz -> SDL-${PV}.tar.gz"
+
+LICENSE="LGPL-2.1"
+SLOT="0"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~mips ~ppc ~ppc64 ~sparc ~x86 ~amd64-linux ~x86-linux"
+# WARNING:
+# If you turn on the custom-cflags use flag in USE and something breaks,
+# you pick up the pieces.  Be prepared for bug reports to be marked INVALID.
+IUSE="oss alsa nas X dga xv xinerama fbcon tslib aalib opengl libcaca +sound +video +joystick custom-cflags pulseaudio static-libs"
```
Comment (a17r) [inline media-libs/libsdl/libsdl-1.2.15_p20210224.ebuild:18]: please sort those

File: media-libs/libsdl/libsdl-1.2.15_p20210224.ebuild [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+	opengl? (
+		>=virtual/opengl-7.0-r1[${MULTILIB_USEDEP}]
+		>=virtual/glu-9.0-r1[${MULTILIB_USEDEP}]
+	)
+	tslib? ( >=x11-libs/tslib-1.0-r3[${MULTILIB_USEDEP}] )
+	pulseaudio? ( >=media-sound/pulseaudio-2.1-r1[${MULTILIB_USEDEP}] )"
+DEPEND="${RDEPEND}
+	nas? ( x11-base/xorg-proto )
+	X? ( x11-base/xorg-proto )
+	x86? ( || ( >=dev-lang/yasm-0.6.0 >=dev-lang/nasm-0.98.39-r3 ) )"
```
Comment (a17r) [inline media-libs/libsdl/libsdl-1.2.15_p20210224.ebuild:45]: DEPENDs/RDEPENDs should also be sorted - in this case all optional, so by IUSE flag, then inside the blocks as well ofc

**Review by a17r — CHANGES_REQUESTED** [review]

File: media-libs/libsdl/libsdl-1.2.15_p20210224.ebuild (hunk 1 above)
Comment (vaukai) [reply media-libs/libsdl/libsdl-1.2.15_p20210224.ebuild]: Done

File: media-libs/libsdl/libsdl-1.2.15_p20210224.ebuild (hunk 2 above)
Comment (vaukai) [reply media-libs/libsdl/libsdl-1.2.15_p20210224.ebuild]: Adjusted

File: media-libs/libsdl/libsdl-1.2.15_p20210224.ebuild (hunk 1 above)
Comment (vaukai) [reply media-libs/libsdl/libsdl-1.2.15_p20210224.ebuild]: Removed multilib and toolchain-funcs and it works without them.

File: media-libs/libsdl/libsdl-1.2.15_p20210224.ebuild (hunk 3 above)
Comment (vaukai) [reply media-libs/libsdl/libsdl-1.2.15_p20210224.ebuild]: Sorted

File: media-libs/libsdl/libsdl-1.2.15_p20210224.ebuild [hunk 5]
```diff
@@ -0,0 +1,135 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit autotools flag-o-matic  multilib-minimal
```
Comment (a17r) [inline media-libs/libsdl/libsdl-1.2.15_p20210224.ebuild:6]: There's a double space in here now

File: media-libs/libsdl/libsdl-1.2.15_p20210224.ebuild [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+	sound? ( >=media-libs/audiofile-0.3.5[${MULTILIB_USEDEP}] )
+	X? (
+		>=x11-libs/libX11-1.6.2[${MULTILIB_USEDEP}]
+		>=x11-libs/libXext-1.3.2[${MULTILIB_USEDEP}]
+		>=x11-libs/libXrandr-1.4.2[${MULTILIB_USEDEP}]
+	)"
+DEPEND="${RDEPEND}
+	nas? ( x11-base/xorg-proto )
+	X? ( x11-base/xorg-proto )
+	x86? ( || ( >=dev-lang/yasm-0.6.0 >=dev-lang/nasm-0.98.39-r3 ) )"
```
Comment (a17r) [inline media-libs/libsdl/libsdl-1.2.15_p20210224.ebuild:46]: Please break it up into one line per dependency

**Review by a17r — CHANGES_REQUESTED** [review]

File: media-libs/libsdl/libsdl-1.2.15_p20210224.ebuild (hunk 5 above)
Comment (vaukai) [reply media-libs/libsdl/libsdl-1.2.15_p20210224.ebuild]: Removed

File: media-libs/libsdl/libsdl-1.2.15_p20210224.ebuild (hunk 6 above)
Comment (vaukai) [reply media-libs/libsdl/libsdl-1.2.15_p20210224.ebuild]: Done

**Comment by chewi** [comment]
This looks good but what's the story with the resizing patch? Why is it not needed?

**Comment by vaukai** [comment]
> This looks good but what's the story with the resizing patch? Why is it not needed?

Actually I cannot see why it is/was needed at all. 

And it fails if I tried to keep it:
` * Applying libsdl-1.2.15-resizing.patch ...`
`patching file src/video/x11/SDL_x11events.c`
`Hunk #2 FAILED at 819.`
`1 out of 2 hunks FAILED -- saving rejects to file src/video/x11/SDL_x11events.c.rej`

Reason for failing is that the coding to be removed has changed.
Applying that patch with Hunk #2 commented out leads to compilation errors.

The package seems to work without that patch.  But as I don't know the error being fixed with it I am asking.  Is it worth reworking this patch?

**Comment by vaukai** [comment]
> This looks good but what's the story with the resizing patch? Why is it not needed?

That patch is from Bug-Debian: http://bugs.debian.org/665779 referring to http://bugzilla.libsdl.org/show_bug.cgi?id=1430 where it says it's fixed upstream with https://hg.libsdl.org/SDL/rev/20cbaf8965fb.  This fix is included in the snapshot.  So I will remove that line from the ebuild.

**Pull request closed without merging**